src/Files.App/Actions/Content/Archives/Compress/CompressIntoArchiveAction.cs

```diff
@@ -32,18 +32,16 @@ public override async Task ExecuteAsync()
 			if (!dialog.CanCreate || result != ContentDialogResult.Primary)
 				return;
 
-			IArchiveCreator creator = new ArchiveCreator
-			{
-				Sources = sources,
-				Directory = directory,
-				FileName = dialog.FileName,
-				Password = dialog.Password,
-				FileFormat = dialog.FileFormat,
-				CompressionLevel = dialog.CompressionLevel,
-				SplittingSize = dialog.SplittingSize,
-			};
+			ICompressArchiveModel creator = new CompressArchiveModel(
+				sources,
+				directory,
+				dialog.FileName,
+				dialog.Password,
+				dialog.FileFormat,
+				dialog.CompressionLevel,
+				dialog.SplittingSize);
```

Comment:
The general change. I need them to be constructor-initialization to provide dev with syntax and list of what they need to fill.